documents/coredocument.go

```diff
@@ -115,6 +117,28 @@ func NewCoreDocumentWithCollaborators(documentPrefix []byte, collaborators Colla
 	return cd, nil
 }
 
+// NewCoreDocumentWithAccessToken generates a new core document with a document type specified by the prefix.
+// It also adds the targetID as a read collaborator, and adds an access token on this document for the document specified in the documentID parameter
+func NewCoreDocumentWithAccessToken(ctx context.Context, documentPrefix []byte, params documentpb.AccessTokenParams) (*CoreDocument, error) {
+	did, err := identity.StringsToDIDs(params.Grantee)
+	if err != nil {
+		return nil, err
+	}
+	collaborators := &CollaboratorsAccess{
+		ReadCollaborators:      []identity.DID{*did[0]},
+		ReadWriteCollaborators: nil,
+	}
+	cd, err := NewCoreDocumentWithCollaborators(documentPrefix, *collaborators)
+	if err != nil {
+		return nil, err
+	}
+	cd, err = cd.AddAccessToken(ctx, params)
```

Comment:
You cant Add the token here like this. This will trigger a new Core Document.

Reply:
AddAccessToken triggers a new version of the core document, not a new core document.